src/inmanta/data/__init__.py

```diff
@@ -604,6 +722,195 @@ def _get_value(cls, value: Any) -> Any:
 
         return value
 
+    @classmethod
+    def get_composed_filter_with_query_types(
```

Comment:
I start feeling that a QueryBuilder is becoming a good idea (even if just to handle the offset magic) .